### PR TITLE
[PLAT-673] Before running nbsmoke lint, copy notebooks to temp folder

### PR DESCRIPTION
**Pull request by looi**
Avoids issue with imports

### CODE REVIEW

File: pre_commit_hooks/nbsmoke_lint.py [hunk 1]
```diff
@@ -11,19 +14,29 @@ def main(argv=None):
...
-            "nbsmoke_flakes_to_ignore=undefined name 'spark'",
-            # Perform nbsmoke lint check.
-            "--nbsmoke-lint",
-        ]
-        + args.filenames
-    )
+    # Copy all notebooks to a temporary directory for the lint check.
+    # Without copying, pytest will try to import the notebook folders as python modules
+    # which is unnecessary and may fail if we don't have the required dependencies.
+    with tempfile.TemporaryDirectory() as tmp_dir:
```
Comment (devstein) [inline pre_commit_hooks/nbsmoke_lint.py:20]: Why does copying to a new directory change the behavior?

**Review by devstein — APPROVED** [review]
Code-wise LGTM, but one no-blocking question

**Review by nikicc — APPROVED** [review]
I've tested and this fixes the issue of `ModuleNotFoundError: No module named 'airflow'` 🚀 💪

File: pre_commit_hooks/nbsmoke_lint.py (hunk 1 above)
Comment (looi) [reply pre_commit_hooks/nbsmoke_lint.py:20]: Yeah, copying to a new folder means `__init__.py` won't exist, so the file won't get run when pytest imports the folder as a python module

**Pull request merged**